### ISSUE TITLE
go failed to build on ArchLinux: internal PATH gets altered incorrectly

### ISSUE THREAD

**Issue by jokeyrhyme** [Linuxbrew/homebrew-core#3119]
# Please always follow these steps:
- [X] Confirmed this is a problem with `brew install`ing one, specific formula and not every time you run `brew`? If it's a general `brew` problem please file this issue at https://github.com/Linuxbrew/brew/issues/new.
- [X] Attempt to replicate the problem on macOS with Homebrew if possible. If you can replicate the problem on macOS with Homebrew, please report the issue to [Homebrew/core](https://github.com/Homebrew/homebrew-core). If you are unsure, please report the issue to Linuxbrew.
- [X] Ran `brew update` and retried your prior step?
- [X] Ran `brew doctor`, fixed all issues and retried your prior step?
- [x] Ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] If `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

**Please note we will close your issue without comment if you delete or do not fill out the issue checklist and provide ALL the requested information.**

`brew gist-logs go`: https://gist.github.com/anonymous/cece6eee3687af9c349114b27b529c08

To help us debug your issue please explain:
## What you were trying to do (and why)

-   trying to `brew install go`, although what I've found makes me think this could be a non-Formula-specific issue

## What happened (include command output)

```
$ brew install go
==> Downloading https://storage.googleapis.com/golang/go1.8.3.src.tar.gz
Already downloaded: /home/ron/.cache/Homebrew/go-1.8.3.tar.gz
==> Downloading https://storage.googleapis.com/golang/go1.7.linux-amd64.tar.gz
Already downloaded: /home/ron/.cache/Homebrew/go--gobootstrap-1.7.tar.gz
==> ./make.bash --no-clean
Last 15 lines from /home/ron/.cache/Homebrew/Logs/go/01.make.bash:
2017-07-05 05:58:42 +1000

./make.bash
--no-clean

##### Building Go bootstrap tool.
cmd/dist

go tool dist: cannot invoke C compiler "gcc-7": exec: "gcc-7": executable file not found in $PATH

Go needs a system C compiler for use with cgo.
To set a C compiler, set CC=the-compiler.
To disable cgo, set CGO_ENABLED=0.


READ THIS: https://github.com/Linuxbrew/brew/blob/master/docs/Troubleshooting.md#troubleshooting
Please do not report this issue to Homebrew/brew or Homebrew/core, which support macOS only.

These open issues may also help:
go 1.8.3 - failure with 'ldd' and 'line-clang.elf' https://github.com/Linuxbrew/homebrew-core/issues/2798
```

## What you expected to happen

-   compiled and installed `go`

## Step-by-step reproduction instructions (by running `brew install` commands)

```sh
brew install go
```

## What I've found

`brew gist-logs go`: https://gist.github.com/anonymous/cece6eee3687af9c349114b27b529c08

```
$ which gcc-7
/home/ron/.linuxbrew/bin/gcc-7
```

In the Gist, take a look at 01.make.bash. You'll notice that `/home/ron/.linuxbrew/bin` is no where in the logged PATH at all

```
$ env | grep ENV
WINDOWPATH=2
CAML_LD_LIBRARY_PATH=/home/ron/.opam/4.03.0/lib/stublibs
PATH=/home/ron/.rbenv/shims:/home/ron/.rbenv/bin:/home/ron/.rbenv/shims:/home/ron/.rbenv/bin:/home/ron/.gem/ruby/2.4.0/bin:/usr/lib/ruby/gems/2.4.0/bin:/home/ron/.opam/4.03.0/bin:/home/ron/.nvs/default/bin:/home/ron/.linuxbrew/bin:/home/ron/go/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/var/lib/snapd/snap/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/ron/Android/Sdk/platform-tools:/home/ron/Android/Sdk/tools
OCAML_TOPLEVEL_PATH=/home/ron/.opam/4.03.0/lib/toplevel
MOZ_PLUGIN_PATH=/usr/lib/mozilla/plugins
INFOPATH=/home/ron/.linuxbrew/share/info:/home/ron/.linuxbrew/share/info:
GOPATH=/home/ron/go
MANPATH=/home/ron/.opam/4.03.0/man:/home/ron/.linuxbrew/share/man:
```

Yet, I can see that `/home/ron/.linuxbrew/bin` is in PATH in my shell session

Oddly enough, changing MAKEFLAGS in my shell session seems to show that variables in my shell session _are_ generally used as-is during `brew install go`, but for some reason PATH is reduced / truncated / altered in a way

**Comment by sjackman** [comment]
Consider installing Linuxbrew in `/home/linuxbrew/.linuxbrew/` if possible so that you can use precompiled binary packages (known as bottles) for non-relocatable formula like `go`.
Another possible workaround for you is `brew install --force-bottle go`, but no promises.

The precompiled binary bottles of non-relocatable bottles can only be used if you install in `/home/linuxbrew/.linuxbrew`, otherwise they have to be built from source. See the documentation below, but wherever it reads `/usr/local`, which is for macOS, read instead `/home/linuxbrew/.linuxbrew`.
+ http://docs.brew.sh/Installation.html#alternative-installs
+ http://docs.brew.sh/FAQ.html#why-do-you-compile-everything
+ http://docs.brew.sh/Bottles.html#cellar-cellar

**Comment by sjackman** [comment]
Do you get this same error with GCC 5?
```sh
brew install go --cc=gcc-5
```

**Comment by jokeyrhyme** [comment]
@sjackman

> Do you get this same error with GCC 5?

_That_ seems to work, in that it doesn't have the weird PATH issue. It still fails with:

> Error: Failure while executing: ["ldd", "/home/ron/.linuxbrew/Cellar/go/1.8.3/libexec/src/debug/dwarf/testdata/line-clang.elf"] 

But that is a different issue: #2798 

I'll try reinstalling brew in `/home/linuxbrew/.linuxbrew` as suggested so I can benefit from bottles. I prefer to keep brew per-user, but the reality is that I don't share my computers and I don't use multiple accounts.

**Comment by sjackman** [comment]
Weird. I'm not sure why it's failing to build. A PR to fix either this issue or issue #2798 would be appreciated, if you're able to figure out either of them. Otherwise using the precompiled bottle is your best bet.
